src/components/guideLineLayer.ts

```diff
@@ -0,0 +1,179 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Components {
+  export class GuideLineLayer<D> extends Component {
+    public static ORIENTATION_VERTICAL = "vertical";
+    public static ORIENTATION_HORIZONTAL = "horizontal";
+
+    private _orientation: string;
+    private _value: D;
+    private _scale: QuantitativeScale<D>;
+    private _pixelPosition: number;
+    private _scaleUpdateCallback: ScaleCallback<QuantitativeScale<D>>;
+    private _guideLine: d3.Selection<void>;
+    private _primaryProperty = "value";
```

Comment:
We should probably name the instance variable.  Note that your commit messages make some instance of a "mode", maybe "propertyMode"?